source/base/function_lib.cc

```diff
@@ -1939,9 +1939,8 @@ namespace Functions
   template <int dim>
   double
   FourierCosineFunction<dim>::value(const Point<dim>  &p,
-                                    const unsigned int component) const
+                                    [[maybe_unused]]  const unsigned int component) const
```

Comment:
I suspect that you have one space too many here. Did you run the indentation script?